hoomd/md/FIREEnergyMinimizerGPU.cc

```diff
@@ -74,6 +80,8 @@ void FIREEnergyMinimizerGPU::update(uint64_t timestep)
     Scalar tnorm(0.0);
     Scalar wnorm(0.0);
 
+    initializePartialSumArrays();
```

Comment:
Yes, this should fix all the issues I identified in #588. However, reallocating on every timestep is expensive. Make a boolean flag that sets when this needs to be updated and only update on those steps where the flag is true. Use `resize` on each of the partial sum arrays instead of new allocations so that memory is only reallocated when needed.

Reply:
I may be wrong, but after looking through the implementation, I believe `resize` will always allocate new memory, even if the new size is the same as the old size.

Reply:
You could use `GPUVector` instead of `GPUArray` or implement your own conditional to check to resize only when needed.

Reply:
Also, ignore what I said about a boolean flag. I considered using signals and slots to track when reallocation is needed, but it would be much cleaner to just check:
```
if (needed size != current size)
    {
    initializePartialSumArrays();
    }
```